source/extensions/grpc_credentials/example/config.cc

```diff
@@ -0,0 +1,72 @@
+#include "extensions/grpc_credentials/example/config.h"
+
+#include "envoy/api/v2/core/grpc_service.pb.h"
+#include "envoy/grpc/google_grpc_creds.h"
+#include "envoy/registry/registry.h"
+
+#include "common/grpc/google_grpc_creds_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace GrpcCredentials {
+namespace Example {
+
+std::shared_ptr<grpc::ChannelCredentials>
+AccessTokenExampleGrpcCredentialsFactory::getChannelCredentials(
+    const envoy::api::v2::core::GrpcService& grpc_service_config) {
+  const auto& google_grpc = grpc_service_config.google_grpc();
+  std::shared_ptr<grpc::ChannelCredentials> creds =
+      grpc::SslCredentials(grpc::SslCredentialsOptions());
+  if (google_grpc.has_channel_credentials() &&
+      google_grpc.channel_credentials().has_ssl_credentials()) {
+    creds = grpc::SslCredentials(
+        Grpc::BuildSslOptionsFromConfig(google_grpc.channel_credentials().ssl_credentials()));
+  }
+  std::shared_ptr<grpc::CallCredentials> call_creds = nullptr;
+  for (const auto& credential : google_grpc.call_credentials()) {
+    switch (credential.credential_specifier_case()) {
+    case envoy::api::v2::core::GrpcService::GoogleGrpc::CallCredentials::kAccessToken: {
+      if (!credential.access_token().empty()) {
+        std::shared_ptr<grpc::CallCredentials> new_call_creds = grpc::MetadataCredentialsFromPlugin(
+            std::make_unique<StaticHeaderAuthenticator>(credential.access_token()));
+        if (call_creds == nullptr) {
+          call_creds = new_call_creds;
+        } else {
+          call_creds = grpc::CompositeCallCredentials(call_creds, new_call_creds);
```

Comment:
Coverage missing here https://50727-65214191-gh.circle-artifacts.com/0/build/envoy/generated/coverage/coverage.source_extensions_grpc_credentials_example_config.cc.html

Reply:
I'm having trouble coming up with a test for this that is verifiable without completely changing the way this plugin works.  This is more of an example of how to combine credentials than something that is expected to actually be used in its current form though.

Reply:
I added a test that should at least run this code.  Will verify once the coverage report posts.